cmd/ncproxy/run.go

```diff
@@ -40,24 +40,62 @@ var (
 	nodeNetSvcClient *nodeNetSvcConn
 )
 
-var (
-	configPath    = flag.String("config", "", "Path to JSON configuration file.")
-	logDir        = flag.String("log-directory", "", "Directory to write ncproxy logs to. This is just panic logs.")
-	registerSvc   = flag.Bool("register-service", false, "Register ncproxy as a Windows service.")
-	unregisterSvc = flag.Bool("unregister-service", false, "Unregister ncproxy as a Windows service.")
-	runSvc        = flag.Bool("run-service", false, "Run ncproxy as a Windows service.")
-)
-
 func etwCallback(sourceID guid.GUID, state etw.ProviderState, level etw.Level, matchAnyKeyword uint64, matchAllKeyword uint64, filterData uintptr) {
 	if state == etw.ProviderStateCaptureState {
 		stacks := debug.DumpStacks()
 		logrus.WithField("stack", stacks).Info("ncproxy goroutine stack dump")
 	}
 }
 
+func app() *cli.App {
+	app := cli.NewApp()
+	app.Name = "ncproxy"
+	app.Usage = "Network configuration proxy"
+	app.Description = `
+ncproxy is a network daemon designed to facilitate container network setup on a machine. It's
+designed to communicate with several agents and simply acts as the proxy between the one requesting
+operations and the one to carry them out.`
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "config,c",
+			Usage: "Path to the configuration file",
+		},
+		cli.StringFlag{
+			Name:  "log-directory",
+			Usage: "Directory to write ncproxy logs to. This is just panic logs.",
+		},
+		cli.BoolFlag{
+			Name:  "register-service",
+			Usage: "Register ncproxy as a Windows service.",
+		},
+		cli.BoolFlag{
+			Name:  "unregister-service",
+			Usage: "Unregister ncproxy as a Windows service.",
+		},
+		cli.BoolFlag{
+			Name:   "run-service",
+			Hidden: true,
+			Usage:  "Run ncproxy as a Windows service.",
+		},
+	}
+	app.Commands = []cli.Command{
+		configCommand,
+	}
+	app.Action = func(context *cli.Context) error {
+		return run(context)
+	}
+	return app
+}
+
 // Run ncproxy
-func run() error {
-	flag.Parse()
+func run(clicontext *cli.Context) error {
+	var (
+		configPath    = clicontext.GlobalString("config")
+		logDir        = clicontext.GlobalString("log-directory")
+		registerSvc   = clicontext.GlobalBool("register-service")
+		unregisterSvc = clicontext.GlobalBool("unregister-service")
+		runSvc        = clicontext.GlobalBool("run-service")
+	)
```

Comment:
should we make this it's own command? Would make the flags a little cleaner to handle and provide descriptions for.

Reply:
So just running ncproxy with no command would display the help options or? I was thinking to just run the thing `./ncproxy.exe` made the most sense and any extra functionality (just config at the moment) could be commands.

Reply:
I think flag handling would be nicer if we went with another command. I don't think there's any issue with no command just printing the help options.

Reply:
Is your concern just that these flags are global to every command? (even if they don't do anything for the others)

Reply:
The purpose of the PR was to turn this into a urfave/cli app, but it seems like all we did was add a subcommand and keep everything else the way it was. I think we should use the urface/cli app style even with run.

Reply:
If we do change, would you actually want the command to start the ncproxy services as `run` or named something else?  Any internal uses of launching this binary we'll need to communicate to folks also if just `./ncproxy.exe` is no longer the way to start the servers. 

I think if a binary has a main purpose (run a server, do x function) and there's certain functionality that would be nice to group together into the same binary, then the .Action + Subcommands flow feels better to me, but I'm not super tied to either.